File: app/src/main/java/com/pnuema/android/bible/ui/viewholders/NumberSelectionViewHolder.java
package com.pnuema.android.bible.ui.viewholders;

import android.view.View;
import android.widget.TextView;

import com.pnuema.android.bible.R;
import com.pnuema.android.bible.ui.dialogs.NumberSelectionListener;

import androidx.recyclerview.widget.RecyclerView;

public final class NumberSelectionViewHolder extends RecyclerView.ViewHolder {
    private TextView textView;

    public NumberSelectionViewHolder(final View itemView) {
        super(itemView);

        textView = itemView.findViewById(R.id.textNumber);
    }

    public void bind(final int position, final boolean isSelected, final NumberSelectionListener onClickListener) {
        textView.setText(String.valueOf(position));

        textView.setTextAppearance(isSelected ? R.style.BookChapterVerse_NumberText_Selected : R.style.BookChapterVerse_NumberText);
        textView.setBackground(textView.getContext().getDrawable(isSelected ? R.drawable.selected_background_rounded : R.drawable.selectable_background_rounded));

        itemView.setOnClickListener(v -> onClickListener.numberSelected(position));
        textView.setOnClickListener(v -> onClickListener.numberSelected(position));
    }
}

File: app/src/main/java/com/pnuema/bible/android/ui/viewholders/NumberSelectionViewHolder.kt
package com.pnuema.bible.android.ui.viewholders

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.pnuema.bible.android.R
import com.pnuema.bible.android.ui.dialogs.NumberSelectionListener
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.listitem_number.view.*

class NumberSelectionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), LayoutContainer {
    override val containerView: View?
        get() = itemView

    fun bind(position: Int, isSelected: Boolean, onClickListener: NumberSelectionListener) {
        itemView.textNumber.text = position.toString()

        itemView.textNumber.setTextAppearance(if (isSelected) R.style.BookChapterVerse_NumberText_Selected else R.style.BookChapterVerse_NumberText)
        itemView.textNumber.background = itemView.textNumber.context.getDrawable(if (isSelected) R.drawable.selected_background_rounded else R.drawable.selectable_background_rounded)

        itemView.setOnClickListener { onClickListener.numberSelected(position) }
        itemView.textNumber.setOnClickListener { onClickListener.numberSelected(position) }
    }
}
